### PR TITLE
Move languagedetect to vendor

### PR DESCRIPTION
**Pull request by MrPetovan**
This is a redux of #428, see https://github.com/friendica/friendica/pull/3871

### CODE REVIEW

File: langfilter/langfilter.php [hunk 1]
```diff
@@ -29,58 +34,75 @@ function langfilter_uninstall() {
...
-		info( t('Language Filter Settings saved.') . EOL);
+	if ($_POST['langfilter-settings-submit']) {
+		set_pconfig(local_user(), 'langfilter', 'languages', trim($_POST['langfilter_languages']));
+		$enable = ((x($_POST, 'langfilter_enable')) ? intval($_POST['langfilter_enable']) : 0);
+		$disable = 1 - $enable;
+		set_pconfig(local_user(), 'langfilter', 'disable', $disable);
+		$minconfidence = 0 + $_POST['langfilter_minconfidence'];
+		if (!$minconfidence) {
+			$minconfidence = 0;
+		} else if ($minconfidence < 0) {
```
Comment (annando) [inline langfilter/langfilter.php]: Shouldn't this be ```elseif``` instead of ```else if```?

File: langfilter/langfilter.php [hunk 2]
```diff
@@ -89,51 +111,71 @@ function langfilter_addon_settings_post(&$a,&$b) {
...
+		$minlen = 32;
+	}
+	if (strlen($b['item']['body']) < $minlen) {
+		return;
+	}
 
-    $spoken_config = get_pconfig(local_user(),'langfilter','languages');
-    $minconfidence = get_pconfig(local_user(),'langfilter','minconfidence');
+	$spoken_config = get_pconfig(local_user(), 'langfilter', 'languages');
+	$minconfidence = get_pconfig(local_user(), 'langfilter', 'minconfidence');
```
Comment (annando) [inline langfilter/langfilter.php]: Please replace the config functions with the new ones.

File: langfilter/langfilter.php (hunk 1 above)
Comment (MrPetovan) [reply langfilter/langfilter.php]: Indeed, good catch.

File: langfilter/langfilter.php (hunk 2 above)
Comment (MrPetovan) [reply langfilter/langfilter.php]: Will do, thanks for the review.

**Pull request merged**